pkg/vm/stackitem/serialization.go

```diff
@@ -36,8 +36,8 @@ type SerializationContext struct {
 	seen         map[Item]sliceNoPointer
 }
 
-// deserContext is an internal deserialization context.
-type deserContext struct {
+// DeserializationContext is a deserialization context.
+type DeserializationContext struct {
```

Comment:
Looking at the resulting API I think we can leave it hidden. The key difference is that `SerializationContext` can be reused for a set of serialized elements (making serialization a bit more efficient, that's why we have it exported in the first place). `DeserializationContext` is a single-use thing.